api/src/main/java/uk/gov/dhsc/htbhf/claimant/reporting/GoogleAnalyticsClient.java

```diff
@@ -0,0 +1,71 @@
+package uk.gov.dhsc.htbhf.claimant.reporting;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.dhsc.htbhf.claimant.exception.GoogleAnalyticsException;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLEncoder;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.springframework.http.HttpMethod.POST;
+import static org.springframework.http.HttpStatus.OK;
+import static org.springframework.http.MediaType.TEXT_PLAIN;
+
+/**
+ * Service for interacting with google analytics.
+ */
+@Service
+@Slf4j
+public class GoogleAnalyticsClient {
+
+    private static final String REPORT_ENDPOINT = "/collect";
+
+    private final URI googleAnalyticsUri;
+    private final RestTemplate restTemplate;
+
+    public GoogleAnalyticsClient(@Value("${google-analytics.base-uri}") String baseUri,
+                                 RestTemplate restTemplate) throws URISyntaxException {
+        this.googleAnalyticsUri = new URI(baseUri + REPORT_ENDPOINT);
+        this.restTemplate = restTemplate;
+    }
+
+    public void reportEvent(Map<String, String> reportProperties) {
+        String payload = createQueryStringWithUrlEncodedValues(reportProperties);
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(TEXT_PLAIN);
```

Comment:
Would it be worthwhile adding a comment (possibly in the JavaDoc for this method) to say why you needed to set this after all the pain you went through on Friday morning?

Reply:
Added a comment above the line.

Reply:
I'd also explain what Google Analytics needs in terms of a payload to try to explain why its done this way.

Reply:
I've suggested adding a link to the measurement protocol documentation, which might cover it

Reply:
I've left in the comment and added a link in the javadoc for the class.